samtranslator/model/api/http_api_generator.py

```diff
@@ -88,6 +102,71 @@ def _construct_http_api(self):
 
         return http_api
 
+    def _add_cors(self):
+        """
+        Add CORS configuration if CORSConfiguration property is set in SAM.
+        Adds CORS configuration only if DefinitionBody is present and
+        APIGW extension for CORS is not present in the DefinitionBody
+        """
+
+        if not self.cors_configuration:
+            return
+
+        if self.cors_configuration and not self.definition_body:
+            raise InvalidResourceException(
+                self.logical_id, "Cors works only with inline OpenApi specified in " "'DefinitionBody' property"
+            )
+
+        # If cors configuration is set to true add * to the allow origins
+        if isinstance(self.cors_configuration, bool):
+            # if cors config is true add Origins as "'*'"
+            properties = CorsProperties()
+
+        elif is_instrinsic(self.cors_configuration):
```

Comment:
why are we doing this check?

We might want to do string->boolean conversion here (treat "False" as actually being false)

Reply:
I tested giving a parameter and that works for `True` and `False` Strings https://github.com/awslabs/serverless-application-model/pull/1381/files#diff-a1ec29c8e1840e48af6c14421bc3dc82R1-R4